lib/ProductOpener/Tags.pm

```diff
@@ -4937,6 +4937,66 @@ sub cmp_taxonomy_tags_alphabetically ($tagtype, $target_lc, $a, $b) {
 		cmp($translations_to{$tagtype}{$b}{$target_lc} || $translations_to{$tagtype}{$b}{"xx"} || $b);
 }
 
+
+=head2 get_knowledge_content ($tag_type, $tag, $lang_id, $country_code)
+
+Fetch knowledge content as HTML about additive, categories,...
+
+This content is used in knowledge panels.
+
+Content is stored as HTML files in `${lang_dir}/${lang_id}/knowledge_panels/${tag_type}`.
+We first check the existence of a file specific to the country specified by `${country_code}`,
+with a fallback on `world` otherwise. This is useful to have a more specific description for some
+countries compared to the `world` base content.
+
+=head3 Arguments
+
+=head4 $tag_type
+
+The type of the tag (e.g. categories, labels, allergens)
+
+=head4 $tag
+
+The tag we want to match, with language prefix (ex: `en:e255`).
+
+=head4 $lang_id
+
+The user language as a 2-letters code (fr, it,...)
+
+=head4 $country_code
+
+The user country as a 2-letters code (fr, it, ch) or `world`
+
+=head3 Return value
+
+If a content exists for the tag type, tag value, language ID and country code, return the HTML text,
+return undef otherwise. 
+
+=cut
+
+sub get_knowledge_content ($tag_type, $tag, $lang_id, $country_code) {
+	# tag value is normalized:
+	# en:250 -> en_250
+	$tag =~ s/:/_/g;
+
+	my $base_dir = "$lang_dir/$lang_id/knowledge_panels/$tag_type";
+	my $file_path = undef;
+
+	foreach my $target_country_code ($country_code, "world") {
+		$file_path = "$base_dir/$tag" . "_" . "$target_country_code.html";
```

Comment:
You can just declare my $file_path here and remove the my $file_path = undef; outside the loop.

```suggestion
		my $file_path = "$base_dir/$tag" . "_" . "$target_country_code.html";
```

Reply:
Ok, I thought it would be better to explicitely return `undef`, so that readers know it's expected, but I'm making the changes.